flixel/system/FlxAssets.hx

```diff
@@ -62,20 +62,47 @@ class FlxAssets
 	 *
 	 * @param   directory          The directory to scan for files
 	 * @param   subDirectories     Whether to include subdirectories
-	 * @param   include            Regular expression for files to include.
-	 *                             Example: `"\.(jpg|png|gif)$"` will only add files with that extension.
-	 * @param   exclude            Regular expression for files to exclude.
-	 *                             Example: `"/exclude|.ogg$"` Will exclude .ogg files and everything in the exclude folder
+	 * @param   include            A string or `EReg` of files to include.
+	 *                             Example: `"*.jpg|*.png|*.gif"` will only add files with that extension
+	 * @param   exclude            A string or `EReg` of files to exclude.
+	 *                             Example: `"/exclude/*|*.ogg"` will exclude .ogg files and everything in the exclude folder
 	 */
-	public static function buildFileReferences(directory = "assets/", subDirectories = false, ?include:String, ?exclude:String,
+	public static function buildFileReferences(directory = "assets/", subDirectories = false, ?include:haxe.macro.Expr, ?exclude:haxe.macro.Expr,
 			?rename:String->String):Array<haxe.macro.Expr.Field>
 	{
 		#if doc_gen
 		return [];
 		#else
-		var includeReg = include != null ? new EReg(include, "") : null;
-		var excludeReg = exclude != null ? new EReg(exclude, "") : null;
-		return flixel.system.macros.FlxAssetPaths.buildFileReferences(directory, subDirectories, includeReg, excludeReg, rename);
+		function toRegex(expr:haxe.macro.Expr):EReg
+		{
+			switch(expr.expr)
+			{
+				case null | EConst(CIdent("null")):
+					return null;
+				case EConst(CString(s, _)):
+					s = EReg.escape(s);
+					s = StringTools.replace(s, "\\*", ".*");
+					s = StringTools.replace(s, "\\|", "|");
+					return new EReg("^" + s + "$", "");
+				case EConst(CRegexp(r, opt)):
+					return new EReg(r, opt);
+				case EArrayDecl(values): // Backwards compatibility: accept an array of file extensions.
```

Comment:
while providing backward compatibility I think it will mainly serve to confuse future newcomers
if an array of Strings is passed can we instead throw an error and recommend a string or EReg to use instead?